x/snapshot/keeper/keeper.go

```diff
@@ -298,11 +308,16 @@ func (k Keeper) executeSnapshot(ctx sdk.Context, counter int64, keyRequirement t
 	return snapshot, nil
 }
 
-func (k Keeper) setLatestCounter(ctx sdk.Context, counter int64) {
+func (k Keeper) setLatestCounterByRole(ctx sdk.Context, counter int64) {
+}
+
+func (k Keeper) setLatestCounter(ctx sdk.Context, counter int64, keyRole tss.KeyRole) {
 	bz := make([]byte, 8)
 	binary.LittleEndian.PutUint64(bz, uint64(counter))
 
 	ctx.KVStore(k.storeKey).Set([]byte(lastCounterKey), bz)
+
+	ctx.KVStore(k.storeKey).Set([]byte(lastCounterKey+"_"+keyRole.String()), bz)
```

Comment:
This won't work very well with exporting of state. With looking at `bz`, one cannot figure out keyRole without analyzing the string key. See https://github.com/axelarnetwork/axelar-core/pull/1011.